irohad/ametsuchi/impl/postgres_wsv_query.cpp

```diff
@@ -132,14 +126,34 @@ namespace iroha {
       return account;
     }
 
+    nonstd::optional<std::string> PostgresWsvQuery::getAccountDetail(
+        const std::string &account_id, const std::string &detail) {
+      pqxx::result result;
+      try {
+        result = transaction_.exec("SELECT data->>'" + detail + "' FROM account WHERE account_id = "
+                                       + transaction_.quote(account_id) + ";");
+      } catch (const std::exception &e) {
+        log_->error(e.what());
+        return nullopt;
+      }
+      if (result.empty()) {
+        log_->info("Account {} not found", account_id);
+        return nullopt;
+      }
+      auto row = result.at(0);
+      std::string res;
+      row.at(0) >> res;
```

Comment:
are you sure that `.at` will not throw an exception?

Reply:
Yes. If no such key exist, it will return empty line. I just checked it.